src/libs/elektra/kdb.c

```diff
@@ -999,6 +999,29 @@ KDB * kdbOpen (const KeySet * contract, Key * errorKey)
 		goto error;
 	}
 
+	// TODO (atmaxinger): improve
+	bool existingError = keyGetMeta (errorKey, "error") != NULL;
+
+	if (!existingError)
+	{
+		// set a dummy value to block errors
+		// any errors that occur will be converted into warnings
+		keySetMeta (errorKey, "error", "blocked");
+	}
+
+	if (initHooks (handle, elektraKs, handle->modules, contract, errorKey) == -1)
+	{
+		ELEKTRA_SET_INSTALLATION_ERROR (errorKey, "Initializing hooks failed. Please see warning of concrete plugin");
```

Comment:
You need to unblock the errors before setting this. Otherwise, you'll not get an error ;) That's actually another reason to move the blocking into `initHooks`. Then it would already be unblocked at this point.

Reply:
Good catch! 🎣